account_invoice_view_payment/models/account_move.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2017 ForgeFlow S.L.
+#        (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def action_view_payments(self):
+        """
+        This function returns an action that display existing payments of given
+        account moves.
+        When only one found, show the payment immediately.
+        """
+        if self.move_type in ("in_invoice", "in_refund"):
+            action = self.env["ir.actions.act_window"]._for_xml_id(
+                "account.action_account_payments_payable"
+            )
+        else:
+            action = self.env["ir.actions.act_window"]._for_xml_id(
+                "account.action_account_payments"
+            )
+        payment = []
+        for move in self:
+            if move._get_reconciled_payments().id:
+                payment.append(move._get_reconciled_payments().id)
```

Comment:
**issue**: `_get_reconciled_payments()` could return multiple payments, so `.id` could then give a SingletonError

I've tested that `_get_reconciled_payments` is safe to use on a recordset with multiple records, so:

```suggestion
        payments = move._get_reconciled_payments().ids
```

Reply:
Updated using ids